extractor_api.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+func NewExtractorAPI(extractor *Extractor) http.Handler {
+	mux := http.NewServeMux()
+
+	mux.HandleFunc("/healthcheck", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			w.Header().Set("Allow", "GET")
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		w.Write([]byte("OK"))
+	})
+
+	mux.HandleFunc("/extract", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			w.Header().Set("Allow", "POST")
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		postBody := make([]byte, 10000)
+		r.Body.Read(postBody)
+		matchedTermIds := extractor.Extract(string(postBody))
```

Comment:
ok that's a good thought. 

had a quick glance at the underlying library and although it takes a byte slice it looks as though it just iterates over it so it could probably accept a reader instead. Maybe an exercise for a rainy day.. 

This code probably does need to be revisited at least so that it can accept an arbitrary sized document (or one larger than 10kb at least), but I think this is sufficient for experimental purposes at the moment.

I've added a trello card about this to our backlog.